x/incentive/keeper/rewards_test.go

```diff
@@ -0,0 +1,341 @@
+	c := types.NewClaim(addr, c("ukava", 1000000), "bnb", 1)
+	_, found = suite.keeper.GetClaim(suite.ctx, addr, "bnb", 1)
+	suite.False(found)
+	suite.NotPanics(func() {
+		suite.keeper.SetClaim(suite.ctx, c)
+	})
+	testC, found := suite.keeper.GetClaim(suite.ctx, addr, "bnb", 1)
+	suite.True(found)
+	suite.Equal(c, testC)
+	suite.NotPanics(func() {
+		suite.keeper.DeleteClaim(suite.ctx, addr, "bnb", 1)
+	})
+	_, found = suite.keeper.GetClaim(suite.ctx, addr, "bnb", 1)
+	suite.False(found)
+}
+
+func (suite *KeeperTestSuite) TestIterateMethods() {
+	suite.addObjectsToStore() // adds 2 objects of each type to the store
+
+	var rewardPeriods types.RewardPeriods
+	suite.keeper.IterateRewardPeriods(suite.ctx, func(rp types.RewardPeriod) (stop bool) {
+		rewardPeriods = append(rewardPeriods, rp)
+		return false
+	})
+	suite.Equal(2, len(rewardPeriods))
+
+	var claimPeriods types.ClaimPeriods
+	suite.keeper.IterateClaimPeriods(suite.ctx, func(cp types.ClaimPeriod) (stop bool) {
+		claimPeriods = append(claimPeriods, cp)
+		return false
+	})
+	suite.Equal(2, len(claimPeriods))
+
+	var claims types.Claims
+	suite.keeper.IterateClaims(suite.ctx, func(c types.Claim) (stop bool) {
+		claims = append(claims, c)
+		return false
+	})
+	suite.Equal(2, len(claims))
+
+	var genIDs types.GenesisClaimPeriodIDs
+	suite.keeper.IterateClaimPeriodIDKeysAndValues(suite.ctx, func(denom string, id uint64) (stop bool) {
+		genID := types.GenesisClaimPeriodID{Denom: denom, ID: id}
+		genIDs = append(genIDs, genID)
+		return false
+	})
+	suite.Equal(2, len(genIDs))
+}
+
+func (suite *KeeperTestSuite) TestHelperMethods() {
+	rp := types.NewRewardPeriod("bnb", suite.ctx.BlockTime(), suite.ctx.BlockTime().Add(time.Hour*168), c("ukava", 100000000), suite.ctx.BlockTime().Add(time.Hour*168*2), time.Hour*8766)
+	suite.keeper.SetRewardPeriod(suite.ctx, rp)
+	suite.keeper.SetNextClaimPeriodID(suite.ctx, "bnb", 1)
+	suite.NotPanics(func() {
+		suite.keeper.HandleRewardPeriodExpiry(suite.ctx, rp)
+	})
+	_, found := suite.keeper.GetClaimPeriod(suite.ctx, 1, "bnb")
+	suite.True(found)
+
+	addr, _ := sdk.AccAddressFromBech32("kava15qdefkmwswysgg4qxgqpqr35k3m49pkx2jdfnw")
+	c1 := types.NewClaim(addr, c("ukava", 1000000), "bnb", 1)
+	suite.keeper.SetClaim(suite.ctx, c1)
+	suite.NotPanics(func() {
+		suite.keeper.AddToClaim(suite.ctx, addr, "bnb", 1, c("ukava", 1000000))
+	})
+	testC, _ := suite.keeper.GetClaim(suite.ctx, addr, "bnb", 1)
+	suite.Equal(c("ukava", 2000000), testC.Reward)
+
+	suite.NotPanics(func() {
+		suite.keeper.AddToClaim(suite.ctx, addr, "xpr", 1, c("ukava", 1000000))
+	})
+
+	suite.SetupTest()
+	reward := types.NewReward(true, "bnb", c("ukava", 1000000000), time.Hour*7*24, time.Hour*24*365, time.Hour*7*24)
+	suite.NotPanics(func() {
+		suite.keeper.CreateNewRewardPeriod(suite.ctx, reward)
+	})
+	_, found = suite.keeper.GetRewardPeriod(suite.ctx, "bnb")
+	suite.True(found)
+
+}
+
+func (suite *KeeperTestSuite) TestCreateAndDeleteRewardsPeriods() {
```

Comment:
While there are only 2 cases here it should still use table tests. We think that table tests should be used for all tests, even if they only have 1 case, as it itemizes the logic being tested and enables future extensibility.